filament/include/filament/Material.h

```diff
@@ -371,7 +371,8 @@ class UTILS_PUBLIC Material : public FilamentAPI {
     //! Indicates whether an existing parameter is a sampler or not.
     bool isSampler(const char* UTILS_NONNULL name) const noexcept;
 
-
+    //! Returns the material source (ASCII) if exists.
+    const char* UTILS_NULLABLE getSource() const noexcept;
```

Comment:
You need to document clearly the lifetime of the `const char*`. Might be better to return a `std::string_view`.